crates/edr_provider/src/requests/eth/evm.rs

```diff
@@ -53,6 +60,14 @@ pub fn handle_set_next_block_timestamp_request(
     Ok(new_timestamp.to_string())
 }
 
+pub fn handle_snapshot_request(data: &mut ProviderData) -> Result<U64, ProviderError> {
+    let snapshot_id = data.make_snapshot();
+
+    Ok(U64::from(snapshot_id))
+}
+
 fn log_block(_mine_block_result: &MineBlockResult<BlockchainError>) -> Result<(), ProviderError> {
```

Comment:
Is this getting called somewhere now? I'd probably comment it out instead then, because it gives the impression at the call site that it's implemented.

Reply:
It is called, yes.

When it is called, it currently logs an error using `log::error!`. How do you propose we keep track of this unimplemented detail without using `unimplemented!()`, as it would cause runtime errors?

Reply:
I did as you suggested: https://github.com/NomicFoundation/hardhat/pull/4617/commits/2d4ec155dc13b239ba6c2d62739f3e52108498d3